### PR TITLE
Feature - IP Anonymization

### PR DESCRIPTION
**Pull request by oakbani**
🖊  anonymizeIp parsed in project_config.rb
🖊  Param added in commonparams in event_builder.rb
🖊  Unit test cases updated in event_builder_spec.rb and project_spec.rb

### CODE REVIEW

File: lib/optimizely/project_config.rb [hunk 1]
```diff
@@ -86,6 +88,7 @@ def initialize(datafile, logger, error_handler)
       @feature_flags = config.fetch('featureFlags', [])
       @groups = config.fetch('groups', [])
       @project_id = config['projectId']
+      @should_anonymize_ip = config['anonymizeIP']
```
Comment (msohailhussain) [inline lib/optimizely/project_config.rb]: it shouldn't be should.

File: lib/optimizely/event_builder.rb [hunk 2]
```diff
@@ -94,9 +94,10 @@ def get_common_params(user_id, attributes)
             visitor_id: user_id
           }
           ],
-          revision: @config.revision,
-          client_name: CLIENT_ENGINE,
-          client_version: VERSION
+        anonymize_ip: @config.should_anonymize_ip,
+        revision: @config.revision,
+        client_name: CLIENT_ENGINE,
+        client_version: VERSION
```
Comment (msohailhussain) [inline lib/optimizely/event_builder.rb]: plz. indent last 4 lines.

Comment (oakbani) [reply lib/optimizely/event_builder.rb]: They are inline with it's sibling parameters. See above.

File: lib/optimizely/event_builder.rb [hunk 3]
```diff
@@ -94,9 +94,10 @@ def get_common_params(user_id, attributes)
             visitor_id: user_id
           }
           ],
-          revision: @config.revision,
-          client_name: CLIENT_ENGINE,
-          client_version: VERSION
+        anonymize_ip: @config.anonymize_ip,
+        revision: @config.revision,
+        client_name: CLIENT_ENGINE,
+        client_version: VERSION
```
Comment (msohailhussain) [inline lib/optimizely/event_builder.rb]: above 4 should be indented. just verify.

**Pull request closed without merging**